src/quilla/__init__.py

```diff
@@ -79,11 +81,94 @@ def make_parser() -> argparse.ArgumentParser:  # pragma: no cover
         metavar='file',
         help='A file with definitions for the \'Definitions\' context object'
     )
+    parser.add_argument(
+        '-v',
+        '--verbose',
+        action='count',
+        help='Flag to increase the verbosity of the outputs. '
+        'Log outputs are directed to stderr by default.',
+        default=0
+    )
 
     return parser
 
 
-def execute(ctx: Context, json_data: str) -> int:
+def get_early_configs(parser: argparse.ArgumentParser, args: List[str]) -> argparse.Namespace:
+    '''
+    Extracts known configs early on. This will disable help/usage printing and prevent
+    exiting on failure, since it potentially needs to ignore errors to collect the early configs
+
+    Args:
+        parser: A parser instance, such as the one returned by `make parser`
+        args: A list of strings containing the arguments to be consumed
+
+    Returns:
+        A namespace object containing the early configurations for Quilla
+
+    '''
+
+    # Saves functions that need to be disabled for early configs
+    exit_function = parser.exit
+    print_help_fn = parser.print_help
+    print_usage_fn = parser.print_usage
+
+    # "Disable" the above functions
+    do_nothing = lambda *x: None  # noqa: E731
+    parser.exit = do_nothing  # type: ignore
+    parser.print_help = do_nothing  # type: ignore
+    parser.print_usage = do_nothing  # type: ignore
+
+    # Get early args
+    parsed_args, _ = parser.parse_known_intermixed_args(args)
+
+    # Revert the disabled functions to their normal usage
+    parser.exit = exit_function  # type: ignore
+    parser.print_help = print_help_fn  # type: ignore
+    parser.print_usage = print_usage_fn  # type: ignore
+
+    return parsed_args
+
+
+def make_default_logger(early_configs: argparse.Namespace) -> logging.Logger:
+    '''
+    Creates a default logger class for Quilla. All internal logger setup logic
+    is done in this function
+
+    Args:
+        early_configs: The extracted configurations from an early pass-through
+            of the parse_known_intermixed_args function, before plugins run
+
+    Returns:
+        A configured ``Logger``
+    '''
+    logger = logging.getLogger('quilla')
+
+    logger.propagate = False  # Disable the root logger
+
+    formatter = logging.Formatter(
+        fmt='[%(asctime)s] [%(levelname)s] - %(message)s',
+        datefmt='%H:%M:%S'
+    )
+
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
+
+    logger.addHandler(stream_handler)
+
+    if early_configs.verbose >= 2:
```

Comment:
Under which condition will ERROR and CRITICAL levels be set?

Reply:
ERROR and CRITICAL will always be shown, regardless of verbosity level. My thought process is that those are significant enough logging factors that they are better suited to be shown always, since it will help identify to the user what they can fix in their validation if it comes up, or to help the user notify us

Reply:
Actually if user sets logger level to be ERROR and verbose = 2, will he ends up getting DEBUG log level?

Reply:
with the python logging module, if you set a lower debugging level, all the levels above it will be displayed. So, by setting it to DEBUG, all logs of DEBUG, INFO, WARNING, ERROR, and CRITICAL will be displayed. By setting the level to WARNING by default, users will get WARNING, ERROR, and CRITICAL logs, they'll also get INFO by adding `-v` and they'll have every log by doing `-vv` or `-v -v`